packages/big-design/src/components/TableNext/SelectAll/helpers.ts

```diff
@@ -0,0 +1,115 @@
+import { TableExpandable, TableSelectable } from '../types';
+
+export function getTotalSelectedItems<T>(
+  items: T[],
+  selectedItems: TableSelectable['selectedItems'],
+) {
+  return items.reduce((acc, _parentRow, parentRowIndex) => {
+    if (selectedItems[parentRowIndex] !== undefined) {
+      return acc + 1;
+    }
+
+    return acc;
+  }, 0);
+}
+
+export function getChildrenRows<T>(
+  parentRow: T,
+  expandedRowSelector?: TableExpandable<T>['expandedRowSelector'],
+) {
+  const expandedRowMode = expandedRowSelector !== undefined;
+
+  if (!expandedRowMode) {
+    return [];
+  }
+
+  return expandedRowSelector(parentRow) ?? [];
+}
+
+export function areAllInPageSelected<T>(
+  isExpandable: boolean,
+  items: T[],
+  selectedItems: TableSelectable['selectedItems'],
+  expandedRowSelector?: TableExpandable<T>['expandedRowSelector'],
+) {
+  return (
```

Comment:
Hmm I think these `return`s in this file could be tweaked a bit. ATM I think it's hard to follow them because theres a `return` -> `items.every` -> more `return` branches -> nested `childrenRows.every` -> `return` branches

Maybe pulling this into logical helper functions? Or moving some stuff around 🤔

Reply:
That's true, I created a helper function so it can be easier to follow 👍🏼